auth.go

```diff
@@ -27,6 +27,9 @@ const (
 	// available only for the Tarantool Enterprise Edition (EE) with
 	// SSL transport.
 	PapSha256Auth
+
+	// NoAuth is used when no authentication is needed.
+	NoAuth Auth = 1 << 15
```

Comment:
We could just skip authentication in `AuthDialer` if Username = "" or not?

Reply:
It is used for the `ProtocolInfo.Auth` field. Is this a bad idea to specify in the `ProtocolInfo` that no authentication should take place (instead of using `AutoAuth` or the response auth from the server)?

Reply:
Yes, in general this does not apply to the protocol that the server sends to a client. The authentication from the client is the `AuthDialer` deal and it looks like we have all data there.